homeassistant/components/lg_netcast/media_player.py

```diff
@@ -79,18 +109,36 @@ class LgTVDevice(MediaPlayerEntity):
     _attr_device_class = MediaPlayerDeviceClass.TV
     _attr_media_content_type = MediaType.CHANNEL
 
-    def __init__(self, client, name, on_action_script):
+    def __init__(self, client, name, *, unique_id=None):
         """Initialize the LG TV device."""
         self._client = client
         self._name = name
         self._muted = False
-        self._on_action_script = on_action_script
+        self._turn_on = PluggableAction(self.async_write_ha_state)
         self._volume = 0
         self._channel_id = None
         self._channel_name = ""
         self._program_name = ""
         self._sources = {}
         self._source_names = []
+        if unique_id is not None:
+            self._attr_unique_id = unique_id
+            self._attr_device_info = DeviceInfo(
+                identifiers={(DOMAIN, unique_id)},
+                manufacturer=ATTR_MANUFACTURER,
+                name=f"{ATTR_MANUFACTURER} Netcast",
+            )
```

Comment:
We should provide `MODEL_NAME` in the `model` field of device info. In the `name` field we can also provide `MANUFACTURER_NAME + MODEL_NAME` or we can use device `friendly_name`. In case of export, we need to use the name from YAML to avoid breaking changes.

See example: https://github.com/home-assistant/core/blob/e0490a3ade04fdc3fcbcacb16ce56ad308408f31/homeassistant/components/braviatv/entity.py#L24-L28

or 
https://github.com/home-assistant/core/blob/e0490a3ade04fdc3fcbcacb16ce56ad308408f31/homeassistant/components/cast/media_player.py#L291-L295

Reply:
We also should set `_attr_has_entity_name = True` and `_attr_name = None` in the `MediaPlayerEntity`. See https://developers.home-assistant.io/blog/2022/07/10/entity_naming/

Reply:
> We should provide `MODEL_NAME` in the `model` field of device info. In the `name` field we can also provide `MANUFACTURER_NAME + MODEL_NAME` or we can use device `friendly_name`. In case of export, we need to use the name from YAML to avoid breaking changes.

I was already generating a `CONF_NAME` but wasn't exposing that through - so I've now switched that to using just the friendly name and there was already a check to check to use the YAML name, which should now be picked up.



> We also should set `_attr_has_entity_name = True` and `_attr_name = None` in the `MediaPlayerEntity`. See https://developers.home-assistant.io/blog/2022/07/10/entity_naming/

Done